src/main/java/serviceimpl/ReportMakingImpl.java

```diff
@@ -0,0 +1,22 @@
+package serviceimpl;
+
+import java.util.List;
+import model.FruitTransaction;
+import service.OperationStrategy;
+import service.ReportMaking;
+
+public class ReportMakingImpl implements ReportMaking {
+    private final OperationStrategy operationStrategy;
+
+    public ReportMakingImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void makeReport(List<FruitTransaction> fruitTransactionList) {
+        for (FruitTransaction transaction : fruitTransactionList) {
+            operationStrategy.findRightStrategy(transaction.getOperation())
+                    .doOperation(transaction);
+        }
```

Comment:
Do we really do make a report here? It looks like we are performing some operations